x-pack/plugins/security_solution/server/endpoint/routes/resolver/entity.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { RequestHandler } from 'kibana/server';
+import { TypeOf } from '@kbn/config-schema';
+import { validateEntities } from '../../../../common/endpoint/schema/resolver';
+import { ResolverEntityIndex } from '../../../../common/endpoint/types';
+
+/**
+ * This is used to get an 'entity_id' which is an internal-to-Resolver concept, from an `_id`, which
+ * is the artificial ID generated by ES for each document.
+ */
+export function handleEntities(): RequestHandler<unknown, TypeOf<typeof validateEntities.query>> {
+  return async (context, request, response) => {
+    const {
+      query: { _id, indices },
+    } = request;
+
+    /**
+     * A safe type for the response based on the semantics of the query.
+     * We specify _source, asking for `process.entity_id` and we only
+     * accept documents that have it.
+     * Also, we only request 1 document.
+     */
+    interface ExpectedQueryResponse {
+      hits: {
+        hits:
+          | []
+          | [
+              {
+                _source: {
+                  process: {
+                    entity_id: string;
+                  };
+                };
+              }
+            ];
+      };
+    }
+
+    const queryResponse: ExpectedQueryResponse = await context.core.elasticsearch.legacy.client.callAsCurrentUser(
```

Comment:
Should we `try/catch` around this? If ES fails for some reason maybe the server should log and return a 500?

Reply:
Let me look into this.

Reply:
Logs in Kibana:
![image](https://user-images.githubusercontent.com/35559/86018105-df6e6680-b9f2-11ea-9bf0-73aad4dab4ed.png)

And the response was: {"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred."}

Reply:
👍  Sounds good, thanks for looking.